scala/build/maintenance.Dockerfile

```
@@ -116,22 +116,29 @@
+    python3-pip && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
 
-# Clean up
-RUN apt-get clean && rm -rf /var/lib/apt/lists/*
+# Optionally verify installation of Python
+RUN python3.12 --version
 
-# Optionally verify installation
-RUN python3.10 --version && pip3 --version
+# Create and activate a virtual environment
+RUN python3.12 -m venv /opt/venv
```

Hadolint

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)